### PR TITLE
Create `Listing` entity

### LINKED ISSUE #216: Create `Listing` entity
### Description
Create `Listing` entity

Database diagram link - https://dbdiagram.io/d/Airbnb-Models-650b038402bd1c4a5ef1b194

### Requirements
- project should be runnable

### Deliverables
- `Listing` entity, `ListingStatus` enum
- `ListingConfiguration` entity configuration
- `ListingRepository` repository
- `ListingService` foundation service
- `ListingsDbContext` data context
- Fluent Valiation
- Add seed data
- Add Migration
- Register all necessary infrastructure to DI

### PR DESCRIPTION
**Pull request by botirjonabdumannonovv**
Closes #216

### CODE REVIEW

**Review by bunyodkamalov — DISMISSED** [review]

**Review by Xolmuratov02 — APPROVED** [review]

**Review by Xolmuratov02 — APPROVED** [review]

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Entities/Listing.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+public class Listing : SoftDeletedEntity
+{
+    public string Name { get; set; } = default!;
+    
+    public DateOnly BuiltDate { get; set; }
+
+    public Address Address { get; set; } = default!;
+    
+    public Money PricePerNight { get; set; } = default!;
+}
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Entities/Listing.cs:18]: Add comment

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Entities/Money.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using AirBnB.Domain.Enums;
+
+namespace AirBnB.Domain.Entities;
+
+public class Money
+{
+    public decimal Amount { get; set; }
+    
+    public Currency Currency { get; set; }
+}
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Entities/Money.cs:10]: Add comment

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Enums/Currency.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿namespace AirBnB.Domain.Enums;
+
+public enum Currency
+{
+    USD
+}
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Enums/Currency.cs:6]: Add comment

File: src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Validators/ListingValidator.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                RuleFor(listing => listing.Id).NotEqual(Guid.Empty);
+            
+            RuleFor(listing => listing.Name)
+                .NotEmpty().WithMessage("Name is required.")
+                .MaximumLength(128).WithMessage("Name cannot exceed 256 characters.");
+
+            RuleFor(listing => listing.BuiltDate).LessThanOrEqualTo(DateOnly.FromDateTime(DateTime.UtcNow));
+
+            RuleFor(listing => listing.PricePerNight)
+                .NotEmpty().WithMessage("Price is required.");
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Validators/ListingValidator.cs:30]: Remove message

File: src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Entities/Address.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+public class Address
+{
+    public string? City { get; set; }
+    
+    public Guid? CityId { get; set; }
+    
+    public double Latitude { get; set; }
+    
+    public double Longitude { get; set; }
+}
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Domain/Entities/Address.cs:12]: Add comment

File: src/Training.AirBnb.Clone.Backend/AirBnB.Api/Models/DTOs/ListingDto.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    /// Gets or sets the unique identifier for the listing
+    /// </summary>
+    public Guid Id { get; set; }
+
+    /// <summary>
+    /// Gets or sets the title of the listing.
+    /// </summary>
+    public string Name { get; set; } = default!;
+    
+    public DateOnly BuiltDate { get; set; }
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Api/Models/DTOs/ListingDto.cs:20]: Add comment

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by Xolmuratov02 — APPROVED** [review]

**Review by bunyodkamalov — APPROVED** [review]

**Review by Xolmuratov02 — APPROVED** [review]
Honey Code))

**Review by sulton-max — APPROVED** [review]

**Pull request merged**